core/include/ikos/core/domain/memory/value.hpp

```diff
@@ -1162,6 +1164,60 @@ class ValueDomain final
     return updated_cells;
   }
 
+
+  /// \brief Detect Read from Uninitialized Memory
+  bool is_read_uninitialized(MemoryLocationRef base,
+                            const MachineInt& offset,
+                            const MachineInt& size) {
+    CellSetT cells = this->_cells.get(base);
+    if (cells.is_empty()) {
+      return true; // If no cells, certainly uninitialized.
+    }
+    if (!offset.fits<long>()) {
+      // Give up if we cannot manipulate the offset natively.
+      return false;
+    }
+    if (!size.fits<long>()) {
+      // Give up if we cannot manipulate the size natively.
+      return false;
+    }
+
+    // Our strategy is to define a set of byte offsets we are
+    // trying to read. Then remove from the set byte offsets
+    // covered by existing cells. If it becomes empty, then
+    // all bytes are initialized.
+    long loffset = offset.to<long>();
+    long lsize = size.to<long>();
+    std::unordered_set<long> bytes;
+    for (long i = 0; i < lsize; ++i) {
+      long byte_offset = loffset + i; // Worry about overflow?
```

Comment:
We could do a check for overflow just once before the loop, by checking if `offset + size` overflows.
We could use one of these functions: https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html

Reply:
With the z number approach there is no longer an overflow issue.